docs/synchronization.md

```diff
@@ -204,30 +109,6 @@ Examples:
 !!! Warning
     If a Workflow is at the front of the queue and it needs to acquire multiple locks, all other Workflows that also need those same locks will wait. This applies even if the other Workflows only wish to acquire a subset of those locks.
 
-## Legacy
-
-In workflows prior to 3.6 you can only specify one lock in any one workflow or template using either a mutex:
-
-```yaml
-    synchronization:
-      mutex:
-     ...
-```
-
-or a semaphore:
-
-```yaml
-    synchronizaion:
-   semamphore:
-     ...
-```
-
-Specifying both would not work in <3.6, only the semaphore would be used.
```

Comment:
We just need to fix this bug per https://github.com/argoproj/argo-workflows/pull/13358#issuecomment-2250795586 and https://github.com/argoproj/argo-workflows/issues/11859 -- i.e. give a validation error instead of silently ignoring it in 3.5 and prior versions

Reply:
See #13669